modules/ROOT/pages/build-an-https-service.adoc

```diff
@@ -0,0 +1,145 @@
+= Configure an HTTPS Service
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: mule runtime, arm, https, cloudhub
+
+To help you ensure your data confidentiality, you can deploy your Mule app using an HTTPS based service. Learn how to configure:
+
+* An HTTPS service to deploy your application to CloudHub
+* HTTPS services using API Manager Proxies
+
+If you need to deploy your app locally, see xref:tls-configuration.adoc[TLS Configuration].
+
+== Prerequisites
+
+Before you begin, build a service (such as a simple "Hello World" service) and deploy it.
+
+== Configure HTTPS Service
+
+Modify your service to HTTPS to deploy your app to CloudHub:
+
+. Generate a `keystore.jks` file using the JDK `keytool` utility on the command line. You must also specify the hostname on the command line to generate a self-signed certificate. +
+
++
+For example, the following command with hostname `SAN=DNS:localhost,IP:127.0.0.1` creates a `keystore.jks` file:
++
+[source,text,linenums]
+----
+keytool -genkeypair -keystore keystore.jks   -dname "CN=localhost, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown"  -keypass password  -storepass password  -keyalg RSA  -sigalg SHA1withRSA  -keysize 2048  -alias mule  -ext SAN=DNS:localhost,IP:127.0.0.1 -validity 9999
+----
+
+. Add the generated `keystore.jks` file to your Anypoint Studio Mule app folder in `src/main/resources`.
+. Configure the HTTP Listener in your Mule app, specifying:
+
+* Any host IP addresses you want to use with the HTTPS scheme for the value of `host`
+* The `${https.port} variable for the value of `port`
+
+[source,xml,linenums]
+----
+...
+<http:listener-config name="HTTPS_Listener_Configuration" protocol="HTTPS" host="0.0.0.0" port="${https.port}" doc:name="HTTP Listener Configuration">
+     <tls:context>
+         <tls:key-store path="keystore.jks" keyPassword="${keystore.password}" password="password"/>
+     </tls:context>
+ </http:listener-config>
+...
+----
+
+[start=4]
+. Include a `config-ref` reference to the HTTPS global listener configuration.
+
+[source,xml,linenums]
+----
+<flow name="httpsserviceFlow">
+    <http:listener config-ref="HTTPS_Listener_Configuration" path="hello" doc:name="HTTPS_Listener_Configuration"/>
```

Comment:
remove the docs. Or make them be the same in each cases. For example remove the underscores and use spaces.